photon-ml/src/main/scala/com/linkedin/photon/ml/avro/model/ModelProcessingUtils.scala

```diff
@@ -109,7 +118,8 @@ object ModelProcessingUtils {
         val Array(featureShardId) = IOUtils.readStringsFromHDFS(idInfoPath, configuration).toArray
 
         // Load the coefficients
-        val featureNameAndTermToIndexMap = featureShardIdToIndexMapLoader(featureShardId).indexMapForDriver()
+        val indexMapLoaders = featureShardIdToIndexMapLoader.flatMap(_.get(featureShardId))
+        val featureNameAndTermToIndexMap = indexMapLoaders.map(_.indexMapForDriver())
```

Comment:
I have mixed feelings about this one, in particular because it makes the variable names pretty long. In some places in the code, the type name is carried exactly by the variable name, resulting in e.g. ```featureShardIdToFeatureNameAndTermToIndexMapMap```. This is of course an extreme case, but it stems from the same idea, that the variable name should provide type information. Not sure what is best here.